config/viper.go

```diff
@@ -1,33 +1,38 @@
 package config
 
 import (
-	"fmt"
 	"log"
 
 	"os"
 	"strings"
 
 	"github.com/bitcoin-sv/pulse/config/p2pconfig"
 	"github.com/bitcoin-sv/pulse/domains/logging"
-	"github.com/bitcoin-sv/pulse/version"
-	"github.com/spf13/pflag"
 	"github.com/spf13/viper"
 )
 
 // Init creates and returns a new viper config.
-func Init(lf logging.LoggerFactory) *Config {
+func Init(lf logging.LoggerFactory) (*Config, *CLI) {
 	viper.AutomaticEnv()
 	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 	setDefaults()
 
+	fs := &PulseFlagSet{}
+	cli := &CLI{}
+
+	initCliFlags(fs, cli)
+	initFlags(fs)
```

Comment:
It's a minor, but maybe we could put `initCliFlags()` inside `initFlags()`

Reply:
And I like idea of two separate functions for this. This makes code more segregated